index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestSIWithRangeColumn.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.spark.testsuite.secondaryindex
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.test.util.QueryTest
+
+class TestSIWithRangeColumn extends QueryTest {
+  test("test SI on range column with and without global sort") {
+    sql("drop table if exists carbon_range_column")
+    sql(
+      """
+        | CREATE TABLE carbon_range_column(id INT, name STRING, city STRING, age INT)
+        | STORED AS carbondata
+        | TBLPROPERTIES(
+        | 'SORT_SCOPE'='LOCAL_SORT', 'SORT_COLUMNS'='name, city', 'range_column'='city')
+      """.stripMargin)
+    sql("CREATE INDEX range_si on carbon_range_column(city) as 'carbondata'")
+    sql("INSERT into carbon_range_column values(1,'nko','blr',25)")
+    checkAnswer(sql("SELECT count(*) FROM range_si"), Seq(Row(1)))
+    checkAnswer(sql("SELECT name FROM carbon_range_column where city='blr'"), Seq(Row("nko")))
+    sql("drop index if exists range_si on carbon_range_column")
+    sql("CREATE INDEX range_si on carbon_range_column(city) as 'carbondata'" +
+      " PROPERTIES('sort_scope'='global_sort', 'Global_sort_partitions'='1')")
+    checkAnswer(sql("SELECT count(*) FROM range_si"), Seq(Row(1)))
+    sql("drop table if exists carbon_range_column")
+  }
+
+  test("test SI creation with range column") {
+    sql("drop table if exists carbon_range_column")
+    sql(
```

Comment:
can extract to new method and reuse create table

Reply:
done